### PR TITLE
fix: transition

### PR DESCRIPTION
**Pull request by s0nicboOm**
Fix transition function for moving models from one stage to another.

### CODE REVIEW

**Review by ab93 — CHANGES_REQUESTED** [review]
Let's do a small change. Move the part of the code that deletes the stale models to a separate private function

**Comment by s0nicboOm** [comment]
Testing the changes on local mlflow. Will let you know when we are good to merge.

**Comment by s0nicboOm** [comment]
Tested successfully on local mlflow.

File: numalogic/registry/mlflow_registry.py [hunk 1]
```diff
@@ -258,16 +263,11 @@ def transition_stage(
...
-            _LOGGER.error("Error when getting model version: %r", ex)
-            return None
+    def __delete_stale_models(self, skeys: Sequence[str], dkeys: Sequence[str]):
+        model_name = self.construct_key(skeys, dkeys)
+        list_model_versions = list(self.client.search_model_versions(f"name='{model_name}'"))
+        if len(list_model_versions) > self.models_to_retain:
+            models_to_delete = list_model_versions[self.models_to_retain :]
+            for stale_model in models_to_delete:
+                self.delete(skeys=skeys, dkeys=dkeys, version=stale_model.version)
+            _LOGGER.info("Deleted Stale models")
```
Comment (ab93) [inline numalogic/registry/mlflow_registry.py:273]: Make this a debug log

Comment (ab93) [reply numalogic/registry/mlflow_registry.py:273]: Also log which versions you deleted.

File: numalogic/registry/mlflow_registry.py [hunk 2]
```diff
@@ -226,30 +226,35 @@ def transition_stage(
...
             # only keep "models_to_retain" number of models.
-            list_model_versions = list(self.client.search_model_versions(f"name='{model_name}'"))
-            models_to_delete = list_model_versions[: -self.models_to_retain]
-            for stale_model in models_to_delete:
-                self.delete(skeys=skeys, dkeys=dkeys, version=stale_model.version)
+            self.__delete_stale_models(skeys=skeys, dkeys=dkeys)
+
             _LOGGER.info("Successfully transitioned model to Production stage")
             return latest_model_data
         except Exception as ex:
```
Comment (ab93) [inline numalogic/registry/mlflow_registry.py:260]: Let's only catch `RestException` here. Because anyway we are calling this under a try-except block

**Review by ab93 — COMMENTED** [review]
Minor comments

File: numalogic/registry/mlflow_registry.py (hunk 1 above)
Comment (s0nicboOm) [reply numalogic/registry/mlflow_registry.py:273]: Good idea. Adding this

**Review by ab93 — APPROVED** [review]

**Pull request merged**